activerecord/lib/active_record/connection_adapters/abstract_adapter.rb

```diff
@@ -169,6 +169,10 @@ def connection_retries
         (@config[:connection_retries] || 1).to_i
       end
 
+      def connection_retry_timeout
+        (@config[:connection_retry_timeout] || 5).to_f
+      end
```

Comment:
I just realized that Matthew's change here set the default in the adapter. I think these methods should be moved into the `HashConfig` and the `DatabaseConfig` should implement an empty method that raises NotImplementedError. I'll move the existing ones that weren't added in this PR.

Reply:
Oh nevermind, I forgot at this point we only have the configuration hash, not the db config, so this is fine. 👍🏼